eo-runtime/src/main/java/org/eolang/ExprReduce.java

```diff
@@ -74,71 +67,138 @@
         final String oper,
         final String param,
-        final Class<T> type,
-        final BiFunction<T, T, T> reduction,
-        final Function<T, String> validation
+        final Args<T> arguments
     ) {
-        this.param = param;
-        this.type = type;
-        this.reduction = reduction;
         this.oper = oper;
-        this.validation = validation;
-    }
-
-    /**
-     * Ctor.
-     * @param oper Peration name
-     * @param param Name of parameter with varargs
-     * @param type Type of varargs
-     * @param reduction Reduction operation on consecutive varargs
-     * @checkstyle ParameterNumberCheck (10 lines)
-     */
-    public ExprReduce(
-        final String oper,
-        final String param,
-        final Class<T> type,
-        final BiFunction<T, T, T> reduction
-    ) {
-        this(
-            oper,
-            param,
-            type,
-            reduction,
-            x -> ""
-        );
+        this.param = param;
+        this.arguments = arguments;
     }
-
     @Override
     public Phi get(final Phi rho) {
-        T acc = new Param(rho).strong(this.type);
+        T acc = this.arguments.accumulator(rho);
         final Phi[] args = new Param(rho, this.param).strong(Phi[].class);
         for (int idx = 0; idx < args.length; ++idx) {
             final Object val = new Dataized(args[idx]).take();
-            if (!val.getClass().getCanonicalName().equals(this.type.getCanonicalName())) {
-                throw new ExFailure(
-                    "The %dth argument of '%s' is not a(n) %s: %s",
-                    idx + 1, this.oper, this.type.getSimpleName(), val
-                );
-            }
-            final T typed = this.type.cast(val);
-            final String msg = this.validation.apply(typed);
+            this.arguments.checkType(val, this.oper, idx);
+            final String msg = this.arguments.validate(val);
             if (!msg.isEmpty()) {
                 throw new ExFailure(
                     "The %dth argument of '%s' is invalid: %s",
                     idx + 1, this.oper, msg
                 );
             }
-            acc = this.reduction.apply(acc, typed);
+            acc = this.arguments.reduce(acc, val);
         }
         return new Data.ToPhi(acc);
     }
+
+    /**
+     * Builds a class that stores some attributes of ExprReduce object.
+     * <p/>Definition example:
+     * <code><pre>
+     *     final VarargExpr&lt;Long&gt; st = new Args<>(
+     *         Long.class,
+     *         (acc, x) -> acc / x,
+     *         x -> {
+     *             String msg = "";
+     *             if (x.equals(0.0)) {
+     *                 msg = "division by zero is infinity";
+     *             }
+     *             return msg;
+     *         }
+     *     );
+     * </pre></code>
+     * @param <T> Type of arguments passing to validation and reduction functions
+     * @since 1.0
+     */
+    public static final class Args<T> {
```

Comment:
@potatmen How about `Args<T>` class has only one public method:
```
/**
* First element is phi itself and then varargs.
*/
public List<T> get(Phi phi) {...}
```
`ExprReduce.get()` would use it as follows:
```
acc = this.args.get(rho).stream()
    .reduce(this.reduction);
```